public/js/pimcore/object/versions.js

```diff
@@ -228,17 +228,19 @@ pimcore.object.versions = Class.create({
             }));
         }
 
-        menu.add(new Ext.menu.Item({
-            text: t('delete'),
-            iconCls: "pimcore_icon_delete",
-            handler: this.removeVersion.bind(this, rowIndex, grid)
-        }));
-
-        menu.add(new Ext.menu.Item({
-            text: t('clear_all'),
-            iconCls: "pimcore_icon_delete",
-            handler: this.removeAllVersion.bind(this, rowIndex, grid)
-        }));
+        if (this.object.isAllowed("delete")) {
```

Comment:
LGTM, just wondering if we need to add the permission check on the controller as well, like https://github.com/pimcore/admin-ui-classic-bundle/blob/eb7b66887a0e59fe3976fc6acd2eb566f39b8861/src/Controller/Admin/Asset/AssetController.php#L955
to 
https://github.com/pimcore/admin-ui-classic-bundle/blob/e414e6bbf487fb412bbb398c9c3b699c875760d3/src/Controller/Admin/ElementController.php#L708-L733